lib/Storage/AmazonS3.php

```diff
@@ -260,20 +260,18 @@ public function opendir($path) {
 
 		try {
 			$files = [];
-			$result = $this->getConnection()->getIterator('ListObjects', [
+			$result = $this->getConnection()->getPaginator('ListObjects', [
 				'Bucket' => $this->bucket,
 				'Delimiter' => '/',
 				'Prefix' => $path
-			], ['return_prefixes' => true]);
+			]);
 
-			foreach ($result as $object) {
-				if (isset($object['Key']) && $object['Key'] === $path) {
-					// it's the directory itself, skip
+			foreach ($result->search('[CommonPrefixes[].Prefix, Contents[].Key][]') as $object) {
+				if (is_string($object) && $object === $path) {
```

Comment:
is the `is_string(..)` check needed?
if the `$object` isn't a string (assuming this is possible, I don't know) you'll be feeding the `basename` function with something which isn't a string.

Reply:
> if the $object isn't a string (assuming this is possible, I don't know) you'll be feeding the basename function with something which isn't a string.

I have updated the change. Now the value to `$file` fed from `basename` will be evaluated in the if condition with `is_string`. The reason why I have used `is_string` is, to be more defensive. If I get some different values, I am ignoring them.